Content.Server/GameObjects/Components/Arcade/SpaceVillainArcadeComponent.cs

```diff
@@ -0,0 +1,403 @@
+        {
+            base.Initialize();
+
+            if (UserInterface != null)
+            {
+                UserInterface.OnReceiveMessage += UserInterfaceOnOnReceiveMessage;
+            }
+        }
+
+
+        private void UserInterfaceOnOnReceiveMessage(ServerBoundUserInterfaceMessage serverMsg)
+        {
+            if (!Powered)
+                return;
+
+            if (!(serverMsg.Message is SpaceVillainArcadePlayerActionMessage msg)) return;
+
+            switch (msg.PlayerAction)
+            {
+                case PlayerAction.Attack:
+                    _game?.ExecutePlayerAction(msg.PlayerAction);
+                    break;
+                case PlayerAction.Heal:
+                    _game?.ExecutePlayerAction(msg.PlayerAction);
+                    break;
+                case PlayerAction.Recharge:
+                    _game?.ExecutePlayerAction(msg.PlayerAction);
+                    break;
+                case PlayerAction.NewGame:
+                    EntitySystem.Get<AudioSystem>().PlayFromEntity("/Audio/Effects/Arcade/newgame.ogg", Owner, AudioParams.Default.WithVolume(-4f));
+                    _game = new SpaceVillainGame(this);
+                    UserInterface?.SendMessage(_game.GenerateMetaDataMessage());
+                    break;
+                case PlayerAction.RequestData:
+                    UserInterface?.SendMessage(_game.GenerateMetaDataMessage());
+                    break;
+            }
+        }
+
+        public enum Wires
+        {
+            /// <summary>
+            /// Disables Max Health&Mana for both Enemy and Player.
+            /// </summary>
+            Overflow,
+            /// <summary>
+            /// Makes Player Invincible.
+            /// </summary>
+            PlayerInvincible,
+            /// <summary>
+            /// Makes Enemy Invincible.
+            /// </summary>
+            EnemyInvincible
+        }
+
+        public void RegisterWires(WiresComponent.WiresBuilder builder)
+        {
+            builder.CreateWire(Wires.Overflow);
+            builder.CreateWire(Wires.PlayerInvincible);
+            builder.CreateWire(Wires.EnemyInvincible);
+        }
+
+        public void WiresUpdate(WiresUpdateEventArgs args)
+        {
+            var wire = (Wires) args.Identifier;
+            var value = args.Action != SharedWiresComponent.WiresAction.Mend;
+            switch (wire)
+            {
+                case Wires.Overflow:
+                    _overflowFlag = value;
+                    break;
+                case Wires.PlayerInvincible:
+                    _playerInvincibilityFlag = value;
+                    break;
+                case Wires.EnemyInvincible:
+                    _enemyInvincibilityFlag = value;
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Called when the user wins the game.
+        /// </summary>
+        public void processWin()
```

Comment:
This needs to follow C#'s naming guidelines: `ProcessWin` instead of `processWin`